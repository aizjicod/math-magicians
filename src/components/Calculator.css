#Calculator {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%,-50%);
  width: 50%;
  height: 500px;
}

#result {
  background: #777;
  color: #fff;
  text-align:end;
  padding: 20px;
}

.pad {
  display: grid;
  grid-template-areas: 
  'uility uility uility leftSide'
  'Numpad Numpad Numpad leftSide'
  'Numpad Numpad Numpad leftSide'
  'Numpad Numpad Numpad leftSide';
}

.utility {
  background: #E0E0E0;
  grid-area: uility;
  display: grid;
  grid-template-columns: repeat(3, minmax(25px, 1fr));
  grid-auto-rows: 50px;
}

.leftSide {
  background: #F5913E;
  grid-area: leftSide;
  display: grid;
  grid-template-columns: minmax(25px, 1fr);
  grid-auto-rows: 50px;
}

.numpad {
  background: #E0E0E0;
  grid-area: Numpad;
  display: grid;
  grid-template-columns: repeat(3, minmax(25px, 1fr));
  grid-auto-rows: 50px;
}

.numpad-0 {
  grid-column: 1/3 ;
}

.numpad *,
.leftSide *,
.utility * {
  border: 1px solid #efefef;
  align-self: stretch;
  display: flex;
  align-items: center;
  justify-content: center;
}
