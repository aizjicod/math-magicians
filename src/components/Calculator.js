import React, { useState } from 'react';
import './Calculator.css';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [total, setTotal] = useState({ total: null });
  const [next, setNext] = useState({ next: null });
  const [operation, setOperation] = useState({ operation: null });
  const [err, setErr] = useState({ err: null });
  const handleClick = (e) => {
    const result = calculate({
      total, next, operation, err,
    }, e.target.value);
    setTotal(() => result.total);
    setNext(() => result.next);
    setOperation(() => result.operation);
    setErr(() => result.err);
  };
  return (
    <div id="Calculator">
      <div id="result">
        {next
          || operation
          || total
          || err
          || 0}
      </div>
      <div className="pad">
        <div className="utility">
          <button type="button" onClick={handleClick} value="AC">AC</button>
          <button type="button" onClick={handleClick} value="+/-">+/-</button>
          <button type="button" onClick={handleClick} value="%">%</button>
        </div>
        <div className="numpad">
          <button type="button" onClick={handleClick} value="7">7</button>
          <button type="button" onClick={handleClick} value="8">8</button>
          <button type="button" onClick={handleClick} value="9">9</button>
          <button type="button" onClick={handleClick} value="4">4</button>
          <button type="button" onClick={handleClick} value="5">5</button>
          <button type="button" onClick={handleClick} value="6">6</button>
          <button type="button" onClick={handleClick} value="1">1</button>
          <button type="button" onClick={handleClick} value="2">2</button>
          <button type="button" onClick={handleClick} value="3">3</button>
          <button type="button" onClick={handleClick} value="0" className="numpad-0">0</button>
          <button type="button" onClick={handleClick} value=".">.</button>
        </div>
        <div className="leftSide">
          <button type="button" onClick={handleClick} value="รท">รท</button>
          <button type="button" onClick={handleClick} value="x">x</button>
          <button type="button" onClick={handleClick} value="-">-</button>
          <button type="button" onClick={handleClick} value="+">+</button>
          <button type="button" onClick={handleClick} value="=">=</button>
        </div>
      </div>
    </div>
  );
};

export default Calculator;
