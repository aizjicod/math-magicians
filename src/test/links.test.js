import { render, fireEvent, screen } from '@testing-library/react';
import Calculator from '../components/calculator';

describe('footer component', () => {
  test('testing if theres a 2022 text in the footer', () => {
    render(<Calculator />);
    const calculatorBtn = screen.getByText(/AC/);
    expect(calculatorBtn).toBeInTheDocument();
  });
});

describe('Calculator tests', () => {
  test('Renders Calculator component', () => {
    const tree = render(<Calculator />);
    expect(tree).toMatchSnapshot();
  });

  test('Simulating user interaction', () => {
    render(<Calculator />);
    const result = document.querySelector('#result');
    const key7 = screen.getByText('7');
    const key8 = screen.getByText('8');
    fireEvent.click(key7);
    fireEvent.click(key8);
    expect(result.textContent).toBe('78');
  });

  test('Simulating user interaction', () => {
    render(<Calculator />);

    const result = document.querySelector('#result');
    const key3 = screen.getByText('3');
    const key7 = screen.getByText('7');
    const keyTimes = screen.getByText('x');
    const keyEquals = screen.getByText('=');

    fireEvent.click(key3);
    fireEvent.click(keyTimes);
    fireEvent.click(key7);
    fireEvent.click(keyEquals);

    expect(result.textContent).toBe('21');
  });
});
