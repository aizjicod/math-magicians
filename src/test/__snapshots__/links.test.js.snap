// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator tests Renders Calculator component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="Calculator"
      >
        <div
          id="result"
        >
          0
        </div>
        <div
          class="pad"
        >
          <div
            class="utility"
          >
            <button
              type="button"
              value="AC"
            >
              AC
            </button>
            <button
              type="button"
              value="+/-"
            >
              +/-
            </button>
            <button
              type="button"
              value="%"
            >
              %
            </button>
          </div>
          <div
            class="numpad"
          >
            <button
              type="button"
              value="7"
            >
              7
            </button>
            <button
              type="button"
              value="8"
            >
              8
            </button>
            <button
              type="button"
              value="9"
            >
              9
            </button>
            <button
              type="button"
              value="4"
            >
              4
            </button>
            <button
              type="button"
              value="5"
            >
              5
            </button>
            <button
              type="button"
              value="6"
            >
              6
            </button>
            <button
              type="button"
              value="1"
            >
              1
            </button>
            <button
              type="button"
              value="2"
            >
              2
            </button>
            <button
              type="button"
              value="3"
            >
              3
            </button>
            <button
              class="numpad-0"
              type="button"
              value="0"
            >
              0
            </button>
            <button
              type="button"
              value="."
            >
              .
            </button>
          </div>
          <div
            class="rightSide"
          >
            <button
              type="button"
              value="รท"
            >
              รท
            </button>
            <button
              type="button"
              value="x"
            >
              x
            </button>
            <button
              type="button"
              value="-"
            >
              -
            </button>
            <button
              type="button"
              value="+"
            >
              +
            </button>
            <button
              type="button"
              value="="
            >
              =
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      id="Calculator"
    >
      <div
        id="result"
      >
        0
      </div>
      <div
        class="pad"
      >
        <div
          class="utility"
        >
          <button
            type="button"
            value="AC"
          >
            AC
          </button>
          <button
            type="button"
            value="+/-"
          >
            +/-
          </button>
          <button
            type="button"
            value="%"
          >
            %
          </button>
        </div>
        <div
          class="numpad"
        >
          <button
            type="button"
            value="7"
          >
            7
          </button>
          <button
            type="button"
            value="8"
          >
            8
          </button>
          <button
            type="button"
            value="9"
          >
            9
          </button>
          <button
            type="button"
            value="4"
          >
            4
          </button>
          <button
            type="button"
            value="5"
          >
            5
          </button>
          <button
            type="button"
            value="6"
          >
            6
          </button>
          <button
            type="button"
            value="1"
          >
            1
          </button>
          <button
            type="button"
            value="2"
          >
            2
          </button>
          <button
            type="button"
            value="3"
          >
            3
          </button>
          <button
            class="numpad-0"
            type="button"
            value="0"
          >
            0
          </button>
          <button
            type="button"
            value="."
          >
            .
          </button>
        </div>
        <div
          class="rightSide"
        >
          <button
            type="button"
            value="รท"
          >
            รท
          </button>
          <button
            type="button"
            value="x"
          >
            x
          </button>
          <button
            type="button"
            value="-"
          >
            -
          </button>
          <button
            type="button"
            value="+"
          >
            +
          </button>
          <button
            type="button"
            value="="
          >
            =
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
